CONTRIBUTING.md

```diff
@@ -0,0 +1,367 @@
+  ```
+
+- **Type check**:
+  ```bash
+  pnpm type-check
+  ```
+
+- **Build stubbed packages** (for faster development):
+  ```bash
+  pnpm stub
+  ```
+
+### Making Changes
+
+1. **Create a new branch**:
+   ```bash
+   git checkout -b my-feature-branch
+   ```
+
+2. **Implement your changes**:
+   - For bug fixes, start by reproducing the issue
+   - For new features, consider how it fits into the existing architecture
+   - Maintain type safety with TypeScript
+   - Add or update tests as necessary
+
+3. **Run local verification**:
+   ```bash
+   # Ensure everything builds
+   pnpm build
+   
+   # Check formatting and lint issues
+   pnpm format
+   pnpm lint
+   
+   # Verify type correctness
+   pnpm type-check
+   
+   # Run tests
+   pnpm test
+   ```
+
+4. **Create a changeset** (for changes that need versioning):
+   ```bash
+   pnpm changeset
+   ```
+   - Follow the prompts to select which packages have changed
+   - Choose the appropriate semver increment (patch, minor, major)
+   - Write a concise but descriptive message explaining the changes
+
+### Testing Your Changes
+
+For testing changes to the core functionality:
+
+1. **Use the examples**:
+   ```bash
+   # Run an example to test your changes
+   pnpm --filter @example/changesets run start
+   ```
+
+2. **Create a test-specific example** (if needed):
+   - Add a new directory in the `examples/` folder
+   - Implement a minimal reproduction that uses your new feature/fix
+   - Run it with `pnpm --filter @example/your-example run start`
+
+### Debugging Tips
+
+When encountering issues during development:
+
+1. **Check for errors in the console** - Clack will often output helpful error messages
+2. **Review the API documentation** - Ensure you're using components and functions as intended
+3. **Look at existing examples** - See how similar features are implemented
+4. **Inspect the packages individually** - Sometimes issues are isolated to either `core` or `prompts`
+
+## Pull Request Process
+
+1. **Commit your changes**:
+   ```bash
+   git add .
+   git commit -m "feat: add new awesome feature"
+   ```
+
+2. **Push to your fork**:
+   ```bash
+   git push origin my-feature-branch
+   ```
+
+3. **Create a pull request**:
+   - Go to your fork on GitHub
+   - Click "New pull request"
+   - Select your branch and add a descriptive title
+   - Fill in the PR template with details about your changes
+   - Reference any related issues
+
+4. **Wait for the automated checks**:
+   - GitHub Actions will run tests, type checking, and lint validation
+   - Fix any issues that arise
+
+5. **Address review feedback**:
+   - Make requested changes
+   - Push additional commits to your branch
+   - The PR will update automatically
+
+### PR Previews
+
+If you want to create a preview build of your PR for others to test, add the label `pr preview` to your PR. This will trigger a workflow that builds a preview version and adds instructions for installation as a comment on your PR.
```

Comment:
```suggestion
Clack uses [pkg.pr.new](https://pkg.pr.new) (provided by [bolt.new](https://bolt.new)) to create continuous preview releases of all PRs. This simplifies testing and makes verifying bug fixes easier for our dependents. 

The workflow that builds a preview version and adds instructions for installation as a comment on your PR should run automatically if you have contributed to Clack before. First-time contributors may need to wait until a maintainer manually approves GitHub Actions running on your PR.
```

Reply:
I adapted this from astro's CONTRIBUTING.md [file](https://github.com/withastro/astro/blob/main/CONTRIBUTING.md)